tests/unit/r2i/preprocessing/normalize.cc

```diff
@@ -0,0 +1,468 @@
+  r2i::ImageFormat frame_format;
+};
+
+class NormalizeMock : public r2i::Normalize {
+
+ public:
+  NormalizeMock () {
+    this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+  }
+
+ private:
+  r2i::RuntimeError SetNormalizationParameters () {
+    this->mean_red = MEAN;
+    this->mean_green = MEAN;
+    this->mean_blue = MEAN;
+    this->std_dev_red = STD_DEV;
+    this->std_dev_green = STD_DEV;
+    this->std_dev_blue = STD_DEV;
+    return r2i::RuntimeError();
+  }
+
+};
+}
+
+/* Override constructors to use smaller matrices for testing */
+namespace r2i {
+
+NormalizeFaceNetV1::NormalizeFaceNetV1 () {
+  this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+};
+
+NormalizeInceptionV1::NormalizeInceptionV1 () {
+  this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+};
+
+NormalizeInceptionV3::NormalizeInceptionV3 () {
+  this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+};
+
+NormalizeResnet50V1::NormalizeResnet50V1 () {
+  this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+};
+
+NormalizeTinyyoloV2::NormalizeTinyyoloV2 () {
+  this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+};
+
+NormalizeTinyyoloV3::NormalizeTinyyoloV3 () {
+  this->dimensions.push_back(std::tuple<int, int>(FRAME_WIDTH, FRAME_HEIGHT));
+};
+
+}
+
+TEST_GROUP(Normalize) {
+  r2i::RuntimeError error;
+  std::shared_ptr<r2i::IPreprocessing> preprocessing =
+    std::make_shared<mock::NormalizeMock>();
+  std::shared_ptr<unsigned char> dummy_frame_data;
+  std::shared_ptr<mock::Frame> in_frame = std::make_shared<mock::Frame>();
+  std::shared_ptr<mock::Frame> out_frame = std::make_shared<mock::Frame>();
+
+  void setup() {
+    error.Clean();
+  }
+};
+
+TEST(Normalize, ApplySuccess) {
+  /* Fill frames with dummy data */
+  unsigned char value = 0;
+  dummy_frame_data = std::shared_ptr<unsigned char>(new unsigned char[FRAME_SIZE],
+                     std::default_delete<const unsigned char[]>());
+
+  unsigned char *in_data = dummy_frame_data.get();
+
+  for (unsigned int i = 0; i < FRAME_SIZE; i++) {
+    in_data[i] = value;
+    value++;
+  }
+
+  error = in_frame->Configure(in_data, FRAME_WIDTH, FRAME_HEIGHT,
+                              r2i::ImageFormat::Id::RGB);
+
+  std::shared_ptr<float> out_data = std::shared_ptr<float>
+                                    (new float[FRAME_WIDTH * FRAME_HEIGHT * CHANNELS],
+                                     std::default_delete<float[]>());
+  error = out_frame->Configure (out_data.get(), FRAME_WIDTH,
+                                FRAME_HEIGHT,
+                                r2i::ImageFormat::Id::RGB);
+
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::EOK, error.GetCode());
+
+  /* Check output values are the expected ones */
+  for (unsigned int i = 0; i < FRAME_SIZE; i++) {
+    DOUBLES_EQUAL( reference_matrix[i],
```

Comment:
This test is a little poor. using a mean value of 0 and a std dev of 1 is not a good enough test. If the processing was bypassed, this would still pass, for example.